mlir-tensorrt/executor/lib/Runtime/Backend/Lua/Modules/TensorRT/TensorRTModule.cpp

```diff
@@ -198,6 +198,195 @@ class NvInferExecContextWrapper {
   /// the TRT network.
   std::vector<PinnedMemoryBlock> hostIOBuffers;
 };
+
+/// OutputAllocatorImpl - Implementation of TensorRT's IOutputAllocator
+/// interface. This class manages memory allocation for TensorRT outputs.
+class OutputAllocatorImpl : public nvinfer1::IOutputAllocator {
+public:
+  OutputAllocatorImpl() = default;
+  /// It is the responsibilty of runtime client to release the buffer.
+  ~OutputAllocatorImpl() {}
+
+  // Disable copy and move operations
+  OutputAllocatorImpl(const OutputAllocatorImpl &) = delete;
+  OutputAllocatorImpl &operator=(const OutputAllocatorImpl &) = delete;
+  OutputAllocatorImpl(OutputAllocatorImpl &&) = delete;
+  OutputAllocatorImpl &operator=(OutputAllocatorImpl &&) = delete;
+
+  void setTensorName(const std::string &tensorName) {
+    mTensorName = tensorName;
+  }
+  void setCurrentMemory(void *currentMemory, int64_t size) {
+    mCurrentMemory = currentMemory;
+    mOutputSize = size;
+  }
+
+  // IOutputAllocator interface implementation
+  void *reallocateOutputAsync(const char *tensorName, void *currentMemory,
+                              uint64_t size, uint64_t alignment,
+                              cudaStream_t stream) override {
+    assert(currentMemory == mCurrentMemory && "Output buffer mismatch");
+    assert(tensorName == mTensorName && "Tensor name mismatch");
+    assert(!mReallocateOutputCalled && "Duplicate call to reallocateOutput");
+    mReallocateOutputCalled = true;
+
+    size = std::max(size, static_cast<uint64_t>(1));
+
+    if (size > mOutputSize) {
+      size = roundUp(size, alignment);
+
+      if (mOutputPtr) {
+        cudaFree(mOutputPtr);
+      }
+
+      mOutputPtr = nullptr;
+      mOutputSize = 0;
+
+      void *newMemory;
+      if (cudaMalloc(&newMemory, size) == cudaSuccess) {
+        mOutputPtr = newMemory;
+        mOutputSize = size;
+      }
+      return mOutputPtr;
+    }
+    return mCurrentMemory;
+  }
+
+  void notifyShape(const char *tensorName,
+                   const nvinfer1::Dims &dims) override {
+    assert(mReallocateOutputCalled &&
+           "TensorRT must invoke reallocateOutput first");
+    assert(!mNotifyShapeCalled && "Duplicate call to notifyShape");
+    assert(tensorName == mTensorName && "Tensor name mismatch");
+    mNotifyShapeCalled = true;
+    mOutputDims = dims;
+  }
+
+  // Accessor methods
+  const nvinfer1::Dims &getOutputDims() const { return mOutputDims; }
+  bool isReallocateOutputCalled() const { return mReallocateOutputCalled; }
```

Comment:
Please remove various unused methods. The `mReallocateOutputCalled` seems like a debugging device; there shouldn't be any need to expose it via class' public API?

Reply:
Yes, those methods were for debugging purposes. I will remove them from public API implementation.